source/code/go/src/plugins/oms.go

```diff
@@ -990,11 +992,28 @@ func InitializePlugin(pluginConfPath string, agentVersion string) {
 		} else {
 			Computer = strings.TrimSuffix(ToString(containerHostName), "\n")
 		}
+		// read proxyendpoint if proxy configured
+		ProxyEndpoint = ""
+		proxyConfigPath := pluginConfig["omsproxy_conf_path"]
+		if _, err := os.Stat(proxyConfigPath); err == nil {
+			Log("Reading proxy configuration for Linux from %s", proxyConfigPath)
+			proxyConfig, err := ioutil.ReadFile(proxyConfigPath)
+			if err != nil {
+				message := fmt.Sprintf("Error Reading omsproxy configuration %s\n", err.Error())
+				Log(message)
+				SendException(message)
+				time.Sleep(30 * time.Second)
```

Comment:
Remove sleep. You are just extending the crashlopp back offs by 2 mins for every backoff. Why fatal? If its fatal (meaning nothing works without proxy), how do you expect appinsights telemetry to work without proper proxy being configured ?

Reply:
Good catch. **AI** may not work if we fail to read the proxy config at all.  If ended up failing to read proxy or bad proxy configuration nothing going to work so sleep is not matter at all. let me remove the seelp.